topics/releases.md

```diff
@@ -1,144 +1,98 @@
+Examples of new functionality include new commands, info fields, or configuration parameters.
+
+Upgrading from a previous minor version should be safe, and will not introduce incompatibilities between servers in the cluster.
 
-Upgrading between minor versions does not introduce any application-level
-compatibility issues.
+**NOTE:** Minor releases may include new commands and data types that can introduce incompatibility between servers in the cluster, but users need to opt-in to these features to cause this type of incompatibility.
+For this reason, it is not recommended to run a Valkey cluster with servers running on different minor versions.
+Users should also avoid new features until all servers in the cluster have been upgrades.
 
-Minor releases may include new commands and data types that introduce
-operations-related incompatibilities, including changes in data persistence
-format and replication protocol.
+Commands may also be deprecated in minor versions.
+If a command is deprecated, a replacement command or an alternative to using the command will be defined in the same minor version.
 
 ### Major versions
 
-Major versions introduce new capabilities and significant changes.
+MAJOR versions are released with significant functionality that may break backwards compatibility or alter key performance characteristics.
+Examples of significant functionality include altering the behavior of an existing command, removing previously deprecated commands, changing the default value of configs, and significant refactoring for performance improvements.
 
-Ideally, these don't introduce application-level compatibility issues.
+Upgrading from a previous major version is intended to be safe, but should be approached with caution. 
+You should carefully read the release notes before performing a major version upgrade.
+Major version upgrades do not guarantee backwards compatibility, which means you should always upgrade replicas before upgrading primaries in order to ensure data consistency.
+
+The Valkey community strives to make as few backwards breaking changes as possible.
+When breaking changes are required, we will also strive to provide a way to mitigate the impact without incuring downtime to your application.
 
 ## Release schedule
 
-A new major version is planned for release once a year.
+The Valkey community strives to release a stable major version once a year.
+Stable minor versions are created as needed in between major releases, and we aim to release at least one minor version a year.
 
-Generally, every major release is followed by a minor version after six months.
+### Release candidate
 
-Patches are released as needed to fix high-urgency issues, or once a stable
-version accumulates enough fixes to justify it.
+New minor and major versions of Valkey begin by branching off the `unstable` branch as an initial release candidate branch, which take the form *major.minor.patch-R#*.
```

Comment:
```suggestion
New minor and major versions of Valkey begin by branching off the `unstable` branch as an initial release candidate branch, which take the form *major.minor.patch-RC#*.
```

Reply:
The release candidate *branch* is just *major.minor*.

The release candidate *tags* are with lowercase "rc" IIRC. I don't know if it's clear (in all parts of the world) that '#' is a place-holder for a digit. Can we use N or something instead, like "_X.Y.Z_-**rc**_N_", or by example *8.0.0-rc1*, *8.0.0-rc2*, etc.

The patch number is always 0 in release canditates?